decoder.go

```diff
@@ -21,6 +21,17 @@ var (
 	nodeType reflect.Type = findNodeType()
 )
 
+// Unmarshal accepts a byte slice as input and writes the
+// data to the value pointed to by v.
+func Unmarshal(bs []byte, v interface{}) error {
```

Comment:
Go has some odd conventions where `Unmarshal` is data first, but things like `io.Copy` are destination first. :-\